ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java

```diff
@@ -176,4 +186,33 @@ public static EndpointResponse mapException(final Throwable exception) {
         .build();
   }
 
+  /**
+   * Interface for reporting metrics to a resource. A resource may choose to break things down
+   * arbitrarily, e.g. /query is used for both push and pull queries so we let the resource
+   * determine how to report the metrics.
+   */
+  public interface MetricsCallback {
```

Comment:
nit: since this is used in the old and the new endpoints can we pull it out into its own class? (same for the impl below)

Reply:
It wasn't used in the new endpoint when I wrote this.  I moved them out and now call it from the new endpoint `stream-query` as well.  I still don't call it from `/ws/query` because it's a bit harder to do it in these scenarios, though we report latency and basic count from there.